internal/database/service.go

```diff
@@ -59,6 +60,33 @@ func (s *Neo4jService) ExecuteWriteQuery(ctx context.Context, cypher string, par
 	return res.Records, nil
 }
 
+// GetQueryType prefixes the provided query with EXPLAIN and returns the query type (e.g. 'r' for read, 'w' for write, 'rw' etc.)
+// This allows read-only tools to determine if a query is safe to run in read-only context.
+func (s *Neo4jService) GetQueryType(ctx context.Context, cypher string, params map[string]any, database string) (neo4j.StatementType, error) {
+	if s.driver == nil {
+		err := fmt.Errorf("neo4j driver is not initialized")
+		log.Printf("error in GetQueryType: %v", err)
+		return neo4j.StatementTypeUnknown, err
+	}
+
+	explainedQuery := strings.Join([]string{"EXPLAIN", cypher}, " ")
+	res, err := neo4j.ExecuteQuery(ctx, *s.driver, explainedQuery, params, neo4j.EagerResultTransformer, neo4j.ExecuteQueryWithDatabase(database))
+	if err != nil {
+		wrappedErr := fmt.Errorf("error during GetQueryType: %w", err)
+		log.Printf("error during GetQueryType: %v", wrappedErr)
```

Comment:
Nittiest of nits: Why are the errors all starting with lowercase "error" instead of "Error"?

Reply:
It's a Go thing. This is the reason: https://go.dev/wiki/CodeReviewComments#error-strings

Reply:
That's not what I'm talking about.

I'm talking about this line of code here:

`log.Printf("error during GetQueryType: %v", wrappedErr)`

The log line is beginning with a lowercase word "error".

Reply:
Oh, I guess that's a preference thing (and a linter setting if you want) which affects how you want your logs to appear on the console such as in GCP when you read it.
https://staticcheck.dev/docs/checks/#ST1005

and as Liam mentioned, it's recommended in Go code review guidelines, that's not mandatory, but maybe frowned upon?:)

Reply:
Oh right, yeah I think logs should be capitalised

> This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.